### DISCUSSION TITLE
Twilio Deepgram integration

### DISCUSSION THREAD

**Discussion by Yagnatheja** [deepgram/community#802] [General help] [answered]
Hello,
I am looking to integrate twilio, deepgram and chatgpt using python. The following is my task:
We call an user using Twilio. Whatever the user speaks, we will send it to deepgram for converting speech to text, converted text is sent to chatgpt, reply from chatgpt is sent to twilio, it is converted to voice and sent back to user. If you have any code at least for integrating twilio and deepgram for live streaming that would be helpful.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by nikolawhallon** [comment, marked as answer]
This guide explains how to stream from Twilio to Deepgram for Speech-to-Text: https://deepgram.com/learn/deepgram-twilio-streaming

This guide explains how to stream Deepgram TTS audio to Twilio: https://developers.deepgram.com/docs/twilio-and-deepgram-tts

Put these two parts together, and add an LLM in between, and things should work!
